### PR TITLE
feat(generate:releaseNotes): Add inline links to headings

### PR DESCRIPTION
**Pull request by tylerbutler**
GitHub does not create heading IDs in the Releases UI, so our release note TOC links don't work. This change adds a remarkjs/unist plugin that inserts HTML anchors into the headings.

Also updates the intro message on the release notes issue to include a command that can be used to generate the release notes locally.

[AB#14174](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/14174)

### CODE REVIEW

File: build-tools/packages/build-cli/src/library/markdown.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+import type { Node } from "unist";
+import { visit } from "unist-util-visit";
+
+/**
+ * Using the same instance for all slug generation ensures that no duplicate IDs are generated.
+ */
+const slugger = new GithubSlugger();
+
+/**
+ * A remarkjs/unist plugin that interts HTML anchor nodes before heading text. This is a workaround for GitHub's lack of
```
Comment (alexvy86) [inline build-tools/packages/build-cli/src/library/markdown.ts:17]:
```suggestion
 * A remarkjs/unist plugin that inserts HTML anchor nodes before heading text. This is a workaround for GitHub's lack of
```

File: build-tools/packages/build-cli/src/library/markdown.ts
Comment (alexvy86) [inline build-tools/packages/build-cli/src/library/markdown.ts:1]: Nit: I probably would have put this in the `releaseNotes.ts` file. Unless we already expect other places to need it, it feels like it's not worth being shared code and requiring imports from internal modules. No strong feelings though.

**Review by alexvy86 — APPROVED** [review]
LGTM

File: build-tools/packages/build-cli/src/library/markdown.ts
Comment (tylerbutler) [reply build-tools/packages/build-cli/src/library/markdown.ts:1]: There are more markdown functions coming, and more uses.

**Pull request merged**